submit/controllers/util.py

```diff
@@ -115,3 +119,88 @@ def pre_validate(self, form: Form) -> None:
     def _value(self) -> str:
         data: str = self.data
         return data
+
+
+class SubmissionMixin:
```

Comment:
Here's the approach for incorporating event validation into form validation. Hopefully light enough to be flexible, but provides enough functionality to cut down on boilerplate